lib/ansible_tower_client/job_template.rb

```diff
@@ -10,11 +10,15 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
+      new_limit = options.dup.delete('limit')
+      response = with_temporary_changes(new_limit) do
+        api.post(launch_url, options).body
+      end
+
+      job = JSON.parse(response)
+      raise InvalidOptions.new(self, Error.parse(response)) if job['job'].nil?
```

Comment:
I still don't know why we need this since we are handling this error [here](https://github.com/ManageIQ/ansible_tower_client/blob/master/lib/ansible_tower_client/collection.rb#L27)